dist/gcc-compatible/curve25519-inline.h

```diff
@@ -1,4 +1,5 @@
 #ifdef __GNUC__
+#if !(defined(__aarch64__) || defined(_M_ARM64) || defined(__arm__) || defined(_M_ARM))
```

Comment:
this is the only remark that I have: for the sake of being future-proof, it'd be good to make this:

```
#if defined(__x86_64__) || defined(_M_X64)
```

however, this file is auto-generated, so we can either:

- fix the generator in vale/code/crypto/ecc/curve25519/Inline25519.ml
- understand why this is being included in non-X64 builds... and how this previously used to work -- my theory: it's the include chain EverCrypt_Curve25519.h --> Hacl_Curve25519_64.h --> curve25519-inline.h that it causing trouble, and ideally, curve25519-inline.h would be included exclusively from Hacl_Curve25519_64.c, not Hacl_Curve25519_64.h

the latter is an easy fix, the former just requires a small option in kremlin (`-add-internal-include`) and a one-liner in the root Makefile

which solution would you prefer?

Reply:
also to be noted that I don't understand how this worked previously on our ARM travis CI